File: src/com/colacelli/irclib/connection/listeners/OnKickListener.java
package com.colacelli.irclib.connection.listeners;

import com.colacelli.irclib.actors.Channel;
import com.colacelli.irclib.actors.User;
import com.colacelli.irclib.connection.Connection;

public interface OnKickListener extends Listener {
    Type TYPE = Type.KICK;

    void onKick(Connection connection, User user, Channel channel);

    @Override
    default Type getType() {
        return TYPE;
    }
}

File: src/main/kotlin/com/colacelli/irclib/connection/listeners/OnKickListener.kt
package com.colacelli.irclib.connection.listeners

import com.colacelli.irclib.actors.Channel
import com.colacelli.irclib.actors.User
import com.colacelli.irclib.connection.Connection

interface OnKickListener : Listener {
    fun onKick(connection: Connection, user: User, channel: Channel)

    override val type: Listener.Type
        get() = TYPE

    companion object {
        val TYPE = Listener.Type.KICK
    }
}
